gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/visitor/ExecutionVisitor.java

```diff
@@ -142,18 +142,18 @@ public EntityFetcherResponse visit(OrNode orNode) {
 
   @Override
   public EntityFetcherResponse visit(SelectionNode selectionNode) {
-    EntityFetcherResponse result = selectionNode.getChildNode().acceptVisitor(this);
+    EntityFetcherResponse childNodeResponse = selectionNode.getChildNode().acceptVisitor(this);
 
     // If the result was empty when the filter is non-empty, it means no entities matched the filter
     // and hence no need to do any more follow up calls.
-    if (result.isEmpty()
+    if (childNodeResponse.isEmpty()
         && !Filter.getDefaultInstance().equals(executionContext.getEntitiesRequest().getFilter())) {
       LOG.debug("No results matched the filter so not fetching aggregate/timeseries metrics.");
-      return result;
+      return childNodeResponse;
```

Comment:
no coverage for this line.

Reply:
Added coverage for this case as well now.